tests/tokens/test_ERC1155ownable.py

```diff
@@ -0,0 +1,474 @@
+import pytest
+import brownie
+from brownie import ZERO_ADDRESS, accounts
+
+# ERC1155 ownable, opensea compatible tests
+# @author Dr. Pixel (github: @Doc-Pixel)
+
+# constants - contract deployment
+CONTRACT_NAME = 'TEST 1155'
+CONTRACT_SYMBOL= 'T1155'
+CONTRACT_URI = 'https://mydomain.io/NFTdata/{id}'
+NEW_CONTRACT_URI = 'https://mynewdomain.io/NFTdata/{id}'
+ERC165_INTERFACE_ID = '0x01ffc9a7'
+ERC1155_INTERFACE_ID = '0xd9b67a26'
+ERC1155_INTERFACE_ID_METADATA = '0x0e89341c'
+
+# minting test lists
+mintBatch = [1,2,3,4,5,6,7,8,9,10]
+mintBatch2 = [11,12,13,14,15,16,17,19,19,20]
+minBatchSetOf10 = [1,1,1,1,1,1,1,1,1,1]
+mintConflictBatch = [1,2,3]
+
+##### test fixtures #####
+
+@pytest.fixture
+def NFT_contract(ERC1155ownable, accounts, scope="module", autouse=True):
+    yield accounts[0].deploy(ERC1155ownable, CONTRACT_NAME, CONTRACT_SYMBOL, CONTRACT_URI)
+
+@pytest.fixture(autouse=True)
+def isolation(fn_isolation):
+    pass
+
+@pytest.fixture
+def test_mint(NFT_contract):
+    NFT_contract.mint(accounts[1], 1, 1, '', {"from": accounts[0]})
+    NFT_contract.mint(accounts[1], 2, 1, '', {"from": accounts[0]})
+    NFT_contract.mint(accounts[1], 3, 1, '', {"from": accounts[0]})
+    
+    with brownie.reverts():
+        # mint with non-owner
+        NFT_contract.mint(accounts[1], 4, 1, '', {"from": accounts[3]})
+
+    with brownie.reverts():
+        # mint to zero address
+        NFT_contract.mint(ZERO_ADDRESS, 4, 1, '', {"from": accounts[0]})
+
+
+@pytest.fixture
+def test_mint_batch(NFT_contract):
+    NFT_contract.mintBatch(accounts[1], mintBatch, minBatchSetOf10, '', {"from": accounts[0]})
+    # NFT_contract.mintBatch(accounts[1], mintBatch2, minBatchSetOf10, '', {"from": accounts[0]})
+
+    with brownie.reverts():
+        # mint with non-owner
+        NFT_contract.mintBatch(accounts[1], mintBatch, minBatchSetOf10, '', {"from": accounts[2]})
+
+    with brownie.reverts():
+        # mint to zero addres
+        NFT_contract.mintBatch(ZERO_ADDRESS, mintBatch, minBatchSetOf10, '', {"from": accounts[0]})
+
+    with brownie.reverts():
+        # ids dont match accounts
+        NFT_contract.mintBatch(accounts[1], [1,2,3], [1,1], '', {"from": accounts[0]})
+
+##### tests #####
+
+def test_initial_state(NFT_contract):
+    # Check if the constructor of the contract is set up properly
+    # and the contract is deployed with the desired variables
+
+    # variables set correctly?
+    assert NFT_contract.name() == CONTRACT_NAME
+    assert NFT_contract.symbol() == CONTRACT_SYMBOL
+    assert NFT_contract.uri() == CONTRACT_URI
+
+    # interfaces set up correctly?
+    tx1 = NFT_contract.supportsInterface(ERC165_INTERFACE_ID) 
```

Comment:
I think this call should be a pure function, so it shouldn't need to be called as a transaction like this

Reply:
I'm not entirely sure what you mean, could you clarify a bit please?

Reply:
Thank you for the explanation!
Updated read-only functions to be @view and the supportsInterface to be @view.